src/main/kotlin/uk/gov/gdx/datashare/services/AcquirersService.kt

```diff
@@ -131,6 +150,47 @@ class AcquirersService(
     }
   }
 
+  fun deleteAcquirerSubscription(subscriptionId: UUID): AcquirerSubscription {
+    val now = dateTimeHandler.now()
+    adminActionAlertsService.noticeAction(
+      AdminAction(
+        "Delete acquirer subscription",
+        object {
+          val subscriptionId = subscriptionId
+          val whenDeleted = now
+        },
+      ),
+    )
+    val subscription = acquirerSubscriptionRepository.save(
+      acquirerSubscriptionRepository.findByIdOrNull(subscriptionId)?.copy(
+        whenDeleted = now,
+      ) ?: throw AcquirerSubscriptionNotFoundException("Subscription $subscriptionId not found"),
+    )
+
+    acquirerSubscriptionEnrichmentFieldRepository
+      .deleteAllByAcquirerSubscriptionId(subscriptionId)
+
+    if (subscription.oauthClientId != null) {
+      val otherSubscriptionsWithClient =
+        acquirerSubscriptionRepository.findAllByOauthClientId(subscription.oauthClientId)
+      if (otherSubscriptionsWithClient.isEmpty()) {
+        cognitoService.deleteUserPoolClient(subscription.oauthClientId)
+      }
+      return subscription
+    }
+
+    if (subscription.queueName == null) {
+      throw IllegalStateException("Acquirer does not have a client id or queue name.")
+    }
+
+    val otherSubscriptionsOnQueue = acquirerSubscriptionRepository.findAllByQueueName(subscription.queueName)
+    if (otherSubscriptionsOnQueue.isEmpty()) {
+      outboundEventQueueService.deleteQueue(subscription.queueName)
+      outboundEventQueueService.deleteQueue("${subscription.queueName}-dlq")
+    }
+    return subscription
```

Comment:
I think maybe for readabilities sake could we do something like this
```suggestion

    acquirerSubscriptionEnrichmentFieldRepository
      .deleteAllByAcquirerSubscriptionId(subscriptionId)

    if (subscription.queueName == null && subscription.oauthClientId == null) {
      throw IllegalStateException("Acquirer does not have a client id or queue name.")
    }

    if (subscription.oauthClientId != null) {
      val otherSubscriptionsWithClient =
        acquirerSubscriptionRepository.findAllByOauthClientId(subscription.oauthClientId)
      if (otherSubscriptionsWithClient.isEmpty()) {
        cognitoService.deleteUserPoolClient(subscription.oauthClientId)
      }
    }
    
    if (subscription.queueName != null) {
      val otherSubscriptionsOnQueue = acquirerSubscriptionRepository.findAllByQueueName(subscription.queueName)
      if (otherSubscriptionsOnQueue.isEmpty()) {
        outboundEventQueueService.deleteQueue(subscription.queueName)
        outboundEventQueueService.deleteQueue("${subscription.queueName}-dlq")
      }
    }
    
    return subscription
```